packages/eslint-plugin/src/rules/consistent-type-imports.ts

```diff
@@ -611,7 +611,7 @@ export default createRule<Options, MessageIds>({
         const { namedSpecifiers: valueImportNamedSpecifiers } =
           classifySpecifier(sourceImports.valueImport);
         if (
-          sourceImports.valueOnlyNamedImport ||
+          sourceImports.valueOnlyNamedImport ??
           valueImportNamedSpecifiers.length
```

Comment:
personally - I hate this style of `if` condition.
I think the `||` is much clearer here.

Reply:
Finally getting to this: enabled `ignoreConditionalTests` in the ESLint config and reverted the now-unnecessary changes.